soroban-env-host/src/expiration_ledger_bumps.rs

```diff
@@ -9,4 +9,4 @@ pub struct LedgerBump {
 }
 
 #[derive(Clone, Default)]
-pub struct ExpirationLedgerBumps(pub Vec<LedgerBump>);
+pub struct ExpirationLedgerBumps(pub MeteredVector<LedgerBump>);
```

Comment:
Does this really need to be a metered vec? IIUC we already charge for allocation of the key, why do we need to copy it again (as well as other keys, due to this being immutable)?

Reply:
Even if there is a reason for this to be a metered vec, it should at least store `Rc`, similar to most of the other metered vecs/maps.

Reply:
I think this is a okay candidate for metered vec. However, the fact metered vec is immutable makes it much uglier to work with and easier to make mistake (like one I made below). I've decided to revert back to using a normal `Vec`.